src/controllers/krate/publish.rs

```diff
@@ -195,11 +194,11 @@ pub async fn publish(app: AppState, req: BytesRequest) -> AppResult<Json<GoodCra
                 process_tarball(&pkg_name, &*tarball_bytes, maximums.max_unpack_size)
                     .map_err(tarball_to_app_error)?;
 
-            let rust_version = tarball_info
-                .manifest
-                .package()
-                .rust_version()
-                .map(|rv| rv.deref().to_string());
+            // `unwrap()` is safe here since `process_tarball()` validates that
+            // we only accept manifests with a `package` section and without
+            // inheritance.
```

Comment:
We might want to add a comment on the `process_tarball()` side as well noting that the validation behaviour is depended on elsewhere, just so future developers don't inadvertently remove it in a refactor.

Reply:
looks like this is already covered by the test suite :)